TST: coverage for _commonType()

tylerjereddy opened the pull request:
Test [uncovered path](https://codecov.io/gh/numpy/numpy/src/master/numpy/linalg/linalg.py#L148) in `np.linalg.linalg._commonType()`, where input array is of an unsupported type.

I couldn't think of more unsupported types offhand, but there may be other subclasses of `inexact` smaller than single precision that I've forgotten about.

tylerjereddy committed:
TST: coverage for _commonType()

* test uncovered path in
linalg.linalg._commonType() when an
unsupported inexact scalar array is
used

eric-wieser reviewed (changes requested):
It would be better to hit this code path via a public function - I suspect that calling any linalg function with a float16 argument would be enough.

tylerjereddy commented:
That's not true -- the first random `linalg` function I chose, `np.linalg.lingal.norm()`, worked just fine with `float16`. I'd also argue that a unit test should probe the actual function even if it is private.

If we want a supplemental *functional* test on a public function that's fine too, but we'll have to pick one that uses that code path explicitly.

eric-wieser commented:
`linalg.cholesky` exhibits the behavior you're after

charris commented:
Needs rebase and resolution of the requested changes.

tylerjereddy commented:
It was already revised to include both private and public function testing, but I've just deleted the private test entirely and rebased since there's not much point in starting a larger debate for such a small adjustment.

charris commented:
For the record, I think it is OK to test private functions when they are going to be widely used internally. If nothing else, it helps development and pins down the functionality.

seberg commented:
Not much point in not merging a test. Eric's comment was also addressed. It is true we do not have many unit testing in numpy, but then we have a few things that are not really tested at all ;). I don't think I mind unit tests, although I think we are OK with mostly integration tests.

Thanks Tyler!
